Sources/Edge.swift

```diff
@@ -175,6 +179,21 @@ public class Edge: NSObject, Extension {
         state?.hitQueue.queue(entity: entity)
     }
 
+    /// Handles shared state update events.
+    /// Updates `ImplementationDetails` if shared state event is from `EventHub`.
+    /// - Parameter event: current shared state update event
+    func handleSharedStateUpdate(_ event: Event) {
```

Comment:
Are we expecting the shared state update to always come first or should we attempt to read on bootup too?

Reply:
Hmm.. I think the bigger issue is if Implementation Details is set before starting the hit queue, and here it is not. EdgeState.bootupIfNeeded is always called first as it is called from `readyForEvent`. I'll need to set the Implementation Details there, before EdgeState updates the hit queue with the initial Consent.